frontend/src/user/profiles/UserProfile.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useParams } from 'react-router-dom';
+import { useCookies } from 'react-cookie';
+import { Nav, Container, Loader, Notification } from '@components/';
+import { UserIcon } from '@icons/';
+import { RootState } from '@store/store';
+import { FollowerRecord, FollowingRecord } from '@store/user';
+import {
+  followUser,
+  getUser,
+  getUserFollowers,
+  getUserFollowing,
+  unfollowUser,
+} from '@api/UserApi';
+import UserProfileItem from './UserProfileItem';
+import { setIsError, setIsLoading } from '@store/category';
+
+interface SpecificUser {
+  id?: number;
+  name?: string;
+  user_image?: string;
+  email?: string;
+  followers?: FollowerRecord[];
+  following?: FollowingRecord[];
+}
```

Comment:
May I know why there this should be different?
Why not have a default `User` we can refer?

Reply:
this is because the response for retrieving the followers/following is different miss.

User response looks like:
```
 {
  id: 2;
  name: 'Name here';
  email: 'email@email.com';
  user_image: 'null';
}
```


but the followers/following response looks like:
`{
            "id": 16,
            "user_id": 2,
            "following": [
                {
                    "id": 5,
                    "name": "Name of user that user2 is following",
                    "email": "email@email.com",
                    "user_image": null
                }
            ]
        }`